language-design.md

```diff
@@ -38,17 +64,28 @@ How do the themes of _Growing a Language_ relate to the lab we did this week?
 
 **Response**
 
+The process we did in lab related to both the content and form of Steele’s paper. Content-wise, we were undertaking the process of growing a small language. We were essentially users of this language who saw features that we wanted to add, because we were given so little direction over what to do. Because the files we were given had only very skeletal definitions, they were very easy to grow and change. The language was written in a fairly intuitive manner, enough so that we could easily understand it and expand upon the basic definitions we were given.
+
+Form-wise, it struck us that Steele only ever defined what he really needed, sometimes going out of his way to avoid defining new terms and discussing why he was making these decisions. When we designed new features to add to the sound lab, we needed to be very careful not to bloat the design with tons of unnecessary features. It would be very bad if, for instance, we defined functions for “increase-volume-and-reverse” and “increase-volume-and-swap-halves” and so on for every possible combination of things we could do to the music. Rather, like Steele did, we can avoid bloat by defining general patterns that allow us to design even more features on top of them.
+
+Another interesting similarity is that Steele defined words in two ways: either by directly defining words, or by defining ways in which words can build into other words (such as adding -ing to verbs). Within the lab, we could do something similar with our design: we could add more building blocks (things similar to changing the volume or reversing), or we could add new ways to combine those building blocks. By doing both of these in conjunction, we can make our language as easy as possible for users to grow.
+
 
 
 ---
  
 **Question**
 
-
 In what way is an API a language? 
 
 **Response**
 
+Like a language, an API is at its core a vocabulary of words you can pass into the API, and meanings for each of those words. In both cases, what’s important is how these words are interpreted in the context of the language and how the user understands the corresponding documentation. An API is essentially a “little language.” Writing a call to an API feels like writing a call to a method in any standard language. This is much like one of the criteria we discussed for DSLs: they have to have a language feel to them [Fowler 2010]. The fact that APIs have a language feel to them is good evidence in favor of their being considered as languages on the same level as DSLs.
+
+Just like any language, an API is growable in the way that Steele discussed, namely in his example of one syllable words and additional definitions. Like our third quote in the first question mentions, you can always add additional functionality to an API as more of a need for it becomes apparent. In fact, it may even be easier with an API because there are fewer interactions between user and language to consider. 
```

Comment:
It's interesting that you point out it may be easier with an API to add functionality. It seems that this would be true assuming there was enough structure to know what functionality to add. Do you think some API's would be easier than others? What could be prohibitive to growth?